### PR TITLE
chore: Build docker images with nix

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Problem

If we are to move to nix we need to build images with nix as well

## Solution

Build docker images with nix, we also add static builds because we want to use them in docker

## Related

- #555

### CODE REVIEW

File: nix/overlay.nix [hunk 1]
```diff
@@ -3,6 +3,12 @@ let disableCheck = package: package.overrideAttrs (o: { doCheck = false; });
 in {
   ocaml-ng = builtins.mapAttrs (_: ocamlVersion:
     ocamlVersion.overrideScope' (oself: osuper: rec {
+      lwt_react = osuper.lwt_react.overrideAttrs
+        (o: { nativeBuildInputs = o.nativeBuildInputs ++ [ oself.cppo ]; });
+      utop = osuper.utop.overrideAttrs (o: {
+        # nativeBuildInputs = o.nativeBuildInputs ++ (with oself; [ dune ocaml cppo ]);
```
Comment (d4hines) [inline nix/overlay.nix:9]: Should we remove this commented code?

**Review by d4hines — COMMENTED** [review]
This is great! I just have one question about the commented-out code.

File: nix/overlay.nix (hunk 1 above)
Comment (ulrikstrid) [reply nix/overlay.nix:9]: Yes!

**Review by d4hines — APPROVED** [review]
This is :ok_hand: :raised_hands: 
nixified wailord approves

**Pull request merged**